cli/src/pcluster/api/controllers/cluster_instances_controller.py

```diff
@@ -49,15 +76,26 @@ def describe_cluster_instances(cluster_name, region=None, next_token=None, node_
 
     :rtype: DescribeClusterInstancesResponseContent
     """
-    return DescribeClusterInstancesResponseContent(
-        [
-            EC2Instance(
-                instance_id="id",
-                launch_time=datetime.now(),
-                public_ip_address="1.2.3.4",
-                instance_type="c5.xlarge",
-                state=InstanceState.RUNNING,
-                private_ip_address="1.2.3.4",
-            )
-        ]
+    cluster = Cluster(cluster_name)
+    node_type = api_node_type_to_cluster_node_type(node_type)
+    instances, next_token = cluster.describe_instances(
+        next_token=next_token, node_type=node_type, queue_name=queue_name
     )
+    ec2_instances = []
+    for instance in instances:
+        instance_info = InstanceInfo(instance)
```

Comment:
would it make sense to have describe_instances from AwsClient return a list of InstanceInfo?

Reply:
Done by using ClusterInfo (a subclass of InstanceInfo)